flink-libraries/flink-table/src/main/scala/org/apache/flink/table/calcite/FlinkPlannerImpl.scala

```diff
@@ -100,6 +101,7 @@ class FlinkPlannerImpl(
       assert(validatedSqlNode != null)
       val rexBuilder: RexBuilder = createRexBuilder
       val cluster: RelOptCluster = RelOptCluster.create(planner, rexBuilder)
+      cluster.setMetadataProvider(FlinkDefaultRelMetadataProvider.INSTANCE)
```

Comment:
Because this logic is called in org.apache.calcite.sql2rel.SqlToRelConverter.convertQuery, which like
`RelMetadataQuery.THREAD_PROVIDERS.set(
        JaninoRelMetadataProvider.of(cluster.getMetadataProvider()));`